components/plan-build/bin/hab-plan-build.sh

```diff
@@ -1977,7 +1977,18 @@ do_default_build_config() {
   _do_copy_templates "config"
   _do_copy_templates "config_install"
   if [[ -d "$PLAN_CONTEXT/hooks" ]]; then
-    cp -r "$PLAN_CONTEXT/hooks" "$pkg_prefix"
+    mkdir -p "$pkg_prefix"/hooks
+    for file in "$PLAN_CONTEXT"/hooks/*
+    do
+      # If there are multiple files for the same hook but different extensions
+      # give precedense to .sh
+      # The supervisor does not recognize extensions so all hooks are
+      # copied without extensions
+      target="$pkg_prefix"/hooks/"$(basename "${file%.*}")"
```

Comment:
How does the supervisor handle files in the `hooks` directory that are not one of the available lifecycle hooks? 

Ex: I could see a user wanting a `shared_functions.sh` file to share among their hooks, and the natural place to put that would be in the hooks directory.  It  looks like the supervisor only copies known hooks to the `/hab/svc/<svc_name>/hooks` directory, so we shouldn't have this case out in the wild, but it would be good to confirm that.

Reply:
Thats absolutely true. The Supervisor only copies files to the `/hab/svc/hooks` directory as part of its hooks compilation process. Only hooks defined in the `HookTable` would be eligible to be compiled.